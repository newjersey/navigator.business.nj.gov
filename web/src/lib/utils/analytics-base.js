const sendEvent = (event) => {
  // console.log("Tracking event:", event.event, event);
  window.gtm(event);
};

const userUpdate = (user_update) => {
  window.userData = { ...window.userData, ...user_update };
  sendEvent({ event: "user_update", ...window.userData });
};

let analyticsContext = { calendar_view: undefined };

const updateContext = (analytics_context) => {
  analyticsContext = { ...analyticsContext, ...analytics_context };
};

export default { sendEvent, userUpdate, updateContext, context: analyticsContext };
