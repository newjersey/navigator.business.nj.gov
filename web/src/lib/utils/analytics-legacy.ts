export type LegacyEventCategory =
  | "opportunity_card"
  | "contextual_link"
  | "external_link"
  | "task_primary_call_to_action"
  | "task_status_checkbox"
  | "landing_page_how_it_works"
  | "landing_page_more_support"
  | "roadmap_dashboard"
  | "formation_nudge_button"
  | "formation_date_modal"
  | "tax_registration_nudge_button"
  | "tax_registration_modal"
  | "myNJ_prompt_modal_complete_button"
  | "calendar_date"
  | "go_to_profile_nudge_button"
  | "go_to_profile_nudge"
  | "landing_page_hero_get_started"
  | "landing_page_navbar_register"
  | "landing_page_second_get_started"
  | "landing_page_find_funding"
  | "contextual_sidebar"
  | "contextual_sidebar_close_button"
  | "tooltip"
  | "onboarding_cannabis_question"
  | "onboarding_liquor_question"
  | "onboarding_car_service_question"
  | "onboarding_moving_company_question"
  | "onboarding_logistics_business_question"
  | "onboarding_childcare_business_question"
  | "onboarding_cpa_question"
  | "onboarding_last_step"
  | "onboarding_last_step_save_additional_business_button"
  | "mandatory_field_error"
  | "roadmap_footer_live_chat_link"
  | "report_something_that_is_broken"
  | "share_feedback"
  | "share_calendar_feedback"
  | "tax_calendar_banner_button"
  | "tax_calendar_modal"
  | "profile_save"
  | "profile_formation_date"
  | "profile_location_question"
  | "task_address_form"
  | "guest_snackbar"
  | "task_location_question"
  | "task_tab_start_application"
  | "task_tab_check_status"
  | "task_button_i_already_submitted"
  | "task_status_checklist_edit_button"
  | "cannabis_license_form_microbusiness_question"
  | "cannabis_license_form_priority_status_diversity_checkbox"
  | "cannabis_license_form_priority_status_impact_checkbox"
  | "cannabis_license_form_priority_status_social_equity_checkbox"
  | "cannabis_license_form"
  | "task_business_name_check_availability"
  | "business_formation_legal_structure_modal"
  | "business_formation_registered_agent_identification"
  | "business_formation_registered_agent_manual_name"
  | "business_formation_registered_agent_manual_address"
  | "business_formation_location_question"
  | "business_formation"
  | "business_formation_members"
  | "business_formation_signers"
  | "business_formation_provisions"
  | "business_formation_purpose"
  | "business_formation_name_step_update_name_on_roadmap"
  | "business_formation_business_name_edit"
  | "business_formation_legal_structure_edit"
  | "business_formation_review_amendments_external_link"
  | "business_formation_success_amendments_external_link"
  | "business_formation_name_tab"
  | "business_formation_business_tab"
  | "business_formation_contacts_tab"
  | "business_formation_billing_tab"
  | "business_formation_review_tab"
  | "business_formation_name_step"
  | "business_formation_name_step_continue_button"
  | "business_formation_business_step_continue_button"
  | "business_formation_contacts_step_continue_button"
  | "business_formation_billing_step_continue_button"
  | "business_formation_dba_resolution_step_continue_button"
  | "business_formation_dba_authorization_step_continue_button"
  | "business_formation_biling_step_continue_button"
  | "business_formation_success_screen"
  | "business_formation_search_existing_name_again"
  | "business_formation_dba_name_search_field"
  | "business_formation_dba_resolution_tab"
  | "business_formation_dba_authorization_tab"
  | "mobile_hamburger_icon"
  | "mobile_menu_close_button"
  | "mobile_menu"
  | "profile_back_to_roadmap"
  | "task_back_to_roadmap"
  | "task_mini_roadmap_step"
  | "task_mini_roadmap_task"
  | "roadmap_task_title"
  | "account_name"
  | "account_menu_myNJ_account"
  | "account_menu_my_profile"
  | "roadmap_section"
  | "roadmap_profile_edit_button"
  | "landing_page_navbar_log_in"
  | "onboarding_header_login"
  | "guest_header_login"
  | "guest_toast"
  | "guest_modal"
  | "guest_menu"
  | "guest_header_register"
  | "roadmap_logout_button"
  | "landing_page_hero_log_in"
  | "finish_setup_on_myNewJersey_button"
  | "landing_page"
  | "for_you_card_hide_button"
  | "for_you_card_unhide_button"
  | "skip_to_main_content_button";

export type LegacyEventAction =
  | "click"
  | "scroll"
  | "arrive"
  | "submit"
  | "click_completed"
  | "click_outside"
  | "appears"
  | "mouseover"
  | "appear"
  | "response";

export type LegacyEventLabel =
  | "go_to_opportunity_screen"
  | "view_sidebar"
  | "open_external_website"
  | "selected_not_started_status"
  | "selected_in_progress_status"
  | "show_legal_structure_modal"
  | "selected_completed_status"
  | "view_business_name_availability"
  | "how_it_works_seen"
  | "more_support_seen"
  | "go_to_name_search_step"
  | "progress_to_needs_to_register_phase"
  | "progress_to_formed_and_registered_phase"
  | "progress_to_up_and_running_phase"
  | "show_formation_date_modal"
  | "formation_status_set_to_complete"
  | "tax_registration_status_set_to_complete"
  | "show_tax_registration_date_modal"
  | "go_to_NavigatorAccount_setup"
  | "go_to_date_detail_screen"
  | "go_to_onboarding"
  | "go_to_onboarding "
  | "arrive_from_myNJ_registration"
  | "close_contextual_sidebar"
  | "view_tooltip"
  | "fail_application_not_found"
  | "success_application_found"
  | "view_application_tab"
  | "view_status_tab"
  | "edit_address_form"
  | "view_requirements"
  | "go_to_profile_screen"
  | "go_to_NIC_formation_processing"
  | "error_remain_at_formation"
  | "members_submitted_with_formation"
  | "signers_submitted_with_formation"
  | "provisions_submitted_with_formation"
  | "purpose_submitted_with_formation"
  | "update_name_on_roadmap"
  | "go_to_Treasury_amendments_page"
  | "arrive_on_business_formation_name_step"
  | "arrive_on_business_formation_business_step"
  | "arrive_on_business_formation_contacts_step"
  | "arrive_on_business_formation_billing_step"
  | "arrive_on_business_formation_review_step"
  | "arrive_from_NIC_formation_processing"
  | "refresh_name_search_field"
  | "dba_name_search_field_appears"
  | "arrive_on_business_formation_dba_resolution_step"
  | "arrive_on_business_formation_dba_authorization_step"
  | "open_mobile_menu"
  | "close_mobile_menu"
  | "expand_contract"
  | "go_to_task"
  | "expand_account_menu"
  | "go_to_myNJ_home"
  | "view_roadmap"
  | "log_out"
  | "unlinked_myNJ_account"
  | "get_unlinked_myNJ_account_modal"
  | "conditional_cannabis_license"
  | "annual_cannabis_license"
  | "yes_require_liquor_license"
  | "no_dont_require_liquor_license"
  | "large_size"
  | "taxi_size"
  | "both_sizes"
  | "yes_moving_across_state_lines"
  | "no_not_moving_across_state_lines"
  | "yes_more_than_6_childred"
  | "no_5_or_fewer_children"
  | "yes_i_offer_public_accounting"
  | "no_i_dont_offer_public_accounting"
  | "finish_onboarding"
  | "finish_additional_business_onboarding"
  | "mandatory_field_error"
  | "open_live_chat"
  | "show_tax_calendar_modal"
  | "show_myNJ_registration_prompt_modal"
  | "tax_calendar_modal_validation_error"
  | "tax_calendar_business_does_not_exist"
  | "tax_deadlines_added_to_calendar"
  | "save_profile_changes"
  | "formation_date_changed"
  | "identified_agent_manually"
  | "entered_agent_ID"
  | "location_entered_for_first_time"
  | "submitted_address_form"
  | "name_is_same_as_account_holder"
  | "address_is_same_as_account_holder"
  | "go_to_myNJ_login"
  | "go_to_profile"
  | "no_I_m_a_standard_business"
  | "yes_I_m_a_microbusiness"
  | "impact_zone_business"
  | "diversely_owned_business"
  | "show_feedback_modal"
  | "social_equity_business"
  | "business_exists_but_not_in_Gov2Go"
  | "get_unlinked_myNJ_account"
  | "go_to_myNJ_registration"
  | "hide_card"
  | "unhide_card"
  | "unhide_cards"
  | "skip_to_main_content";
