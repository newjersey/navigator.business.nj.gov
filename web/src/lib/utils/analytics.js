// eslint-disable-next-line no-undef
const GOOGLE_ANALYTICS_ID = process.env.GOOGLE_ANALYTICS_ID || "";

const sendEvent = (category, action, label) => {
  window.gtag("event", action, {
    event_category: category,
    event_label: label,
  });
};

export default {
  pageview: (path) => {
    window.gtag("config", GOOGLE_ANALYTICS_ID, {
      page_path: path,
    });
  },

  dimensions: {
    industry: (value) => {
      window.gtag("set", "dimension1", value);
    },
    municipality: (value) => {
      window.gtag("set", "dimension2", value);
    },
    legalStructure: (value) => {
      window.gtag("set", "dimension3", value);
    },
    liquorLicense: (value) => {
      window.gtag("set", "dimension4", value);
    },
    homeBasedBusiness: (value) => {
      window.gtag("set", "dimension5", value);
    },
    persona: (value) => {
      window.gtag("set", "dimension6", value);
    },
    registrationStatus: (value) => {
      window.gtag("set", "dimension7", value);
    },
    abExperience: (value) => {
      window.gtag("set", "dimension8", value);
    },
    naicsCode: (value) => {
      window.gtag("set", "dimension9", value);
    },
  },

  event: {
    landing_page_hero_log_in: {
      click: {
        go_to_myNJ_login: () => {
          sendEvent("landing_page_hero_log_in", "click", "go_to_myNJ_login");
        },
      },
    },
    landing_page_hero_get_started: {
      click: {
        go_to_onboarding: () => {
          sendEvent("landing_page_hero_get_started", "click", "go_to_onboarding");
        },
      },
    },
    landing_page_navbar_log_in: {
      click: {
        go_to_myNJ_login: () => {
          sendEvent("landing_page_navbar_log_in", "click", "go_to_myNJ_login");
        },
      },
    },
    landing_page_navbar_register: {
      click: {
        go_to_onboarding: () => {
          sendEvent("landing_page_navbar_register", "click", "go_to_onboarding");
        },
      },
    },
    landing_page_second_get_started: {
      click: {
        go_to_onboarding: () => {
          sendEvent("landing_page_second_get_started", "click", "go_to_onboarding");
        },
      },
    },
    landing_page_find_funding: {
      click: {
        go_to_onboarding: () => {
          sendEvent("landing_page_find_funding", "click", "go_to_onboarding");
        },
      },
    },
    guest_toast: {
      click: {
        go_to_myNJ_registration: () => {
          sendEvent("guest_toast", "click", "go_to_myNJ_registration");
        },
      },
    },
    guest_modal: {
      click: {
        go_to_myNJ_registration: () => {
          sendEvent("guest_modal", "click", "go_to_myNJ_registration");
        },
        go_to_myNJ_login: () => {
          sendEvent("guest_modal", "click", "go_to_myNJ_login");
        },
      },
    },
    guest_menu: {
      click: {
        go_to_myNJ_registration: () => {
          sendEvent("guest_menu", "click", "go_to_myNJ_registration");
        },
        go_to_myNJ_login: () => {
          sendEvent("guest_modal", "click", "go_to_myNJ_login");
        },
      },
    },
    roadmap_dashboard: {
      arrive: {
        arrive_from_myNJ_registration: () => {
          sendEvent("roadmap_dashboard", "arrive", "arrive_from_myNJ_registration");
        },
      },
    },
    roadmap_task_title: {
      click: {
        go_to_task: () => {
          sendEvent("roadmap_task_title", "click", "go_to_task");
        },
      },
    },
    roadmap_logout_button: {
      click: {
        log_out: () => {
          sendEvent("roadmap_logout_button", "click", "log_out");
        },
      },
    },
    roadmap_profile_edit_button: {
      click: {
        go_to_profile_screen: () => {
          sendEvent("roadmap_profile_edit_button", "click", "go_to_profile_screen");
        },
      },
    },
    roadmap_PBS_link: {
      click: {
        go_to_PBS: () => {
          sendEvent("roadmap_PBS_link", "click", "go_to_PBS");
        },
      },
    },
    roadmap_footer_live_chat_link: {
      click: {
        open_live_chat: () => {
          sendEvent("roadmap_footer_live_chat_link", "click", "open_live_chat");
        },
      },
    },
    roadmap_section: {
      click: {
        expand_contract: () => {
          sendEvent("roadmap_section", "click", "expand_contract");
        },
      },
    },
    roadmap_graduate_button: {
      click: {
        view_graduation_modal: () => {
          sendEvent("roadmap_graduate_button", "click", "view_graduation_modal");
        },
      },
    },
    graduation_modal: {
      submit: {
        prospective_roadmap_to_existing_dashboard: () => {
          sendEvent("graduation_modal", "submit", "prospective_roadmap_to_existing_dashboard");
        },
      },
    },
    dashboard_ungraduate_button: {
      click: {
        existing_dashboard_to_prospective_roadmap: () => {
          sendEvent("dashboard_ungraduate_button", "click", "existing_dashboard_to_prospective_roadmap");
        },
      },
    },
    profile_save: {
      click: {
        save_profile_changes: () => {
          sendEvent("profile_save", "click", "save_profile_changes");
        },
      },
    },
    profile_back_to_roadmap: {
      click: {
        view_roadmap: () => {
          sendEvent("profile_back_to_roadmap", "click", "view_roadmap");
        },
      },
    },
    task_back_to_roadmap: {
      click: {
        view_roadmap: () => {
          sendEvent("task_back_to_roadmap", "click", "view_roadmap");
        },
      },
    },
    task_mini_roadmap_step: {
      click: {
        expand_contract: () => {
          sendEvent("task_mini_roadmap_step", "click", "expand_contract");
        },
      },
    },
    task_mini_roadmap_task: {
      click: {
        go_to_task: () => {
          sendEvent("task_mini_roadmap_task", "click", "go_to_task");
        },
      },
    },
    task_primary_call_to_action: {
      click: {
        open_external_website: () => {
          sendEvent("task_primary_call_to_action", "click", "open_external_website");
        },
      },
    },
    task_tab_start_application: {
      click: {
        view_application_tab: () => {
          sendEvent("task_tab_start_application", "click", "view_application_tab");
        },
      },
    },
    task_tab_check_status: {
      click: {
        view_status_tab: () => {
          sendEvent("task_tab_check_status", "click", "view_status_tab");
        },
      },
    },
    task_button_i_already_submitted: {
      click: {
        view_status_tab: () => {
          sendEvent("task_button_i_already_submitted", "click", "view_status_tab");
        },
      },
    },
    task_address_form: {
      submit: {
        submitted_address_form: () => {
          sendEvent("task_address_form", "submit", "submitted_address_form");
        },
      },
      response: {
        fail_application_not_found: () => {
          sendEvent("task_address_form", "response", "fail_application_not_found");
        },
        success_application_found: () => {
          sendEvent("task_address_form", "response", "success_application_found");
        },
      },
    },
    task_status_checklist_edit_button: {
      click: {
        edit_address_form: () => {
          sendEvent("task_status_checklist_edit_button", "click", "edit_address_form");
        },
      },
    },
    task_status: {
      click: {
        dropdown_appears: () => {
          sendEvent("task_status", "click", "dropdown_appears");
        },
      },
    },
    task_status_dropdown: {
      click_not_started: {
        selected_not_started_status: () => {
          sendEvent("task_status_dropdown", "click_not_started", "selected_not_started_status");
        },
      },
      click_in_progress: {
        selected_in_progress_status: () => {
          sendEvent("task_status_dropdown", "click_in_progress", "selected_in_progress_status");
        },
      },
      click_completed: {
        selected_completed_status: () => {
          sendEvent("task_status_dropdown", "click_completed", "selected_completed_status");
        },
      },
    },
    contextual_link: {
      click: {
        view_sidebar: () => {
          sendEvent("contextual_link", "click", "view_sidebar");
        },
      },
    },
    tooltip: {
      mouseover: {
        view_tooltip: () => {
          sendEvent("tooltip", "mouseover", "view_tooltip");
        },
      },
    },
    external_link: {
      click: {
        open_external_website: () => {
          sendEvent("external_link", "click", "open_external_website");
        },
      },
    },
    contextual_sidebar: {
      click_outside: {
        close_contextual_sidebar: () => {
          sendEvent("contextual_sidebar", "click_outside", "close_contextual_sidebar");
        },
      },
    },
    contextual_sidebar_close_button: {
      click: {
        close_contextual_sidebar: () => {
          sendEvent("contextual_sidebar_close_button", "click", "close_contextual_sidebar");
        },
      },
    },
    mobile_hamburger_icon: {
      click: {
        open_mobile_menu: () => {
          sendEvent("mobile_hamburger_icon", "click", "open_mobile_menu");
        },
      },
    },
    mobile_menu_close_button: {
      click: {
        close_mobile_menu: () => {
          sendEvent("mobile_menu_close_button", "click", "close_mobile_menu");
        },
      },
    },
    mobile_menu: {
      click_outside: {
        close_mobile_menu: () => {
          sendEvent("mobile_menu", "click_outside", "close_mobile_menu");
        },
      },
    },
    task_business_name_check_availability: {
      submit: {
        view_business_name_availability: () => {
          sendEvent("task_business_name_check_availability", "submit", "view_business_name_availability");
        },
      },
    },
    account_menu_my_profile: {
      click: {
        go_to_profile_screen: () => {
          sendEvent("account_menu_my_profile", "click", "go_to_profile_screen");
        },
      },
    },
    account_menu_myNJ_account: {
      click: {
        go_to_myNJ_home: () => {
          sendEvent("account_menu_myNJ_account", "click", "go_to_myNJ_home");
        },
      },
    },
    account_name: {
      click: {
        expand_account_menu: () => {
          sendEvent("account_name", "click", "expand_account_menu");
        },
      },
    },
    onboarding_last_step: {
      submit: {
        finish_onboarding: () => {
          sendEvent("onboarding_last_step", "submit", "finish_onboarding");
        },
      },
    },
    onboarding_cpa_question: {
      submit: {
        yes_i_offer_public_accounting: () => {
          sendEvent("onboarding_cpa_question", "submit", "yes_i_offer_public_accounting");
        },
        no_i_dont_offer_public_accounting: () => {
          sendEvent("onboarding_cpa_question", "submit", "no_i_dont_offer_public_accounting");
        },
      },
    },
    onboarding_cannabis_question: {
      submit: {
        conditional_cannabis_license: () => {
          sendEvent("onboarding_cannabis_question", "submit", "conditional_cannabis_license");
        },
        annual_cannabis_license: () => {
          sendEvent("onboarding_cannabis_question", "submit", "annual_cannabis_license");
        },
      },
    },
    business_formation: {
      submit: {
        go_to_NIC_formation_processing: () => {
          sendEvent("business_formation", "submit", "go_to_NIC_formation_processing");
        },
        error_remain_at_formation: () => {
          sendEvent("business_formation", "submit", "error_remain_at_formation");
        },
      },
    },
    business_formation_success_screen: {
      arrive: {
        arrive_from_NIC_formation_processing: () => {
          sendEvent("business_formation_success_screen", "arrive", "arrive_from_NIC_formation_processing");
        },
      },
    },
    business_formation_business_name_edit: {
      click: {
        go_to_name_search_step: () => {
          sendEvent("business_formation_business_name_edit", "click", "go_to_name_search_step");
        },
      },
    },
    business_formation_legal_structure_edit: {
      click: {
        show_legal_structure_modal: () => {
          sendEvent("business_formation_legal_structure_edit", "click", "show_legal_structure_modal");
        },
      },
    },
    business_formation_legal_structure_modal: {
      submit: {
        go_to_profile_screen: () => {
          sendEvent("business_formation_legal_structure_edit", "click", "go_to_profile_screen");
        },
      },
    },
    business_formation_business_step_continue_button: {
      click: {
        go_to_next_formation_step: () => {
          sendEvent("business_formation_business_step_continue_button", "click", "go_to_next_formation_step");
        },
      },
    },
    business_formation_contacts_step_continue_button: {
      click: {
        go_to_next_formation_step: () => {
          sendEvent("business_formation_contacts_step_continue_button", "click", "go_to_next_formation_step");
        },
      },
    },
    business_formation_review_step_continue_button: {
      click: {
        go_to_next_formation_step: () => {
          sendEvent("business_formation_review_step_continue_button", "click", "go_to_next_formation_step");
        },
      },
    },
    business_formation_billing_step_continue_button: {
      click: {
        go_to_next_formation_step: () => {
          sendEvent("business_formation_billing_step_continue_button", "click", "go_to_next_formation_step");
        },
      },
    },
    business_formation_registered_agent_identification: {
      submit: {
        entered_agent_ID: () => {
          sendEvent("business_formation_registered_agent_identification", "submit", "entered_agent_ID");
        },
        identified_agent_manually: () => {
          sendEvent(
            "business_formation_registered_agent_identification",
            "submit",
            "identified_agent_manually"
          );
        },
      },
    },
    business_formation_registered_agent_manual_name: {
      submit: {
        name_is_same_as_account_holder: () => {
          sendEvent(
            "business_formation_registered_agent_manual_name",
            "submit",
            "name_is_same_as_account_holder"
          );
        },
      },
    },
    business_formation_registered_agent_manual_address: {
      submit: {
        address_is_same_as_account_holder: () => {
          sendEvent(
            "business_formation_registered_agent_manual_address",
            "submit",
            "address_is_same_as_account_holder"
          );
        },
      },
    },
    business_formation_name_step_continue_button: {
      click: {
        go_to_next_formation_step: () => {
          sendEvent("business_formation_name_step_continue_button", "click", "go_to_next_formation_step");
        },
      },
    },
    cannabis_license_form_microbusiness_question: {
      submit: {
        yes_I_m_a_microbusiness: () => {
          sendEvent("cannabis_license_form_microbusiness_question", "submit", "yes_I_m_a_microbusiness");
        },
        no_I_m_a_standard_business: () => {
          sendEvent("cannabis_license_form_microbusiness_question", "submit", "no_I_m_a_standard_business");
        },
      },
    },
    cannabis_license_form: {
      click: {
        view_requirements: () => {
          sendEvent("cannabis_license_form", "click", "view_requirements");
        },
      },
    },
    cannabis_license_form_priority_status_diversity_checkbox: {
      submit: {
        diversely_owned_business: () => {
          sendEvent(
            "cannabis_license_form_priority_status_diversity_checkbox",
            "submit",
            "diversely_owned_business"
          );
        },
      },
    },
    cannabis_license_form_priority_status_impact_checkbox: {
      submit: {
        impact_zone_business: () => {
          sendEvent(
            "cannabis_license_form_priority_status_impact_checkbox",
            "submit",
            "impact_zone_business"
          );
        },
      },
    },
    cannabis_license_form_priority_status_social_equity_checkbox: {
      submit: {
        social_equity_business: () => {
          sendEvent(
            "cannabis_license_form_priority_status_social_equity_checkbox",
            "submit",
            "social_equity_business"
          );
        },
      },
    },
    calendar_date: {
      click: {
        go_to_date_detail_screen: () => {
          sendEvent("calendar_date", "click", "go_to_date_detail_screen");
        },
      },
    },
  },
};
