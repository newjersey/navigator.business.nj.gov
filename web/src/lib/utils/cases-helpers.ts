export const capitalizeFirstLetter = (text: string): string => {
  return text.charAt(0).toUpperCase() + text.slice(1);
};

export const camelCaseToSentence = (text: string): string => {
  const spacedCase = text
    .split(/(?=[A-Z])/)
    .join(" ")
    .toLowerCase();
  return capitalizeFirstLetter(spacedCase);
};

export const kebabSnakeSentenceToCamelCase = (text: string): string => {
  return text
    .toLowerCase()
    .split(/[\s_-]/gm)
    .map((cased, index) => {
      return index == 0 ? cased : capitalizeFirstLetter(cased);
    })
    .join("");
};

export const camelCaseToKebabCase = (text: string): string => {
  return text
    .split(/(?=[A-Z])/)
    .join("-")
    .toLowerCase();
};
