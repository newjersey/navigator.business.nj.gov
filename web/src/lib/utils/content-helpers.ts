export const convertTextToMarkdownBold = (text: string): string => {
  if (!text) return "";

  // Without trimming, leading/trailing spaces cause underscores to literally be displayed
  const trimmedText = text.trim();

  return `__${trimmedText}__`;
};
