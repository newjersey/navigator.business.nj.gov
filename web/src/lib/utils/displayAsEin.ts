export const displayAsEin = (value: string): string => {
  const numerals = [...value];
  if (numerals.length > 2) {
    numerals.splice(2, 0, "-");
  }
  return numerals.join("");
};
