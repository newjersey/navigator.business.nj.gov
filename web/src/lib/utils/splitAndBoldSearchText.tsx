import { ReactElement } from "react";

export const splitAndBoldSearchText = (displayText: string, searchText: string): ReactElement => {
  const index = displayText.toLowerCase().indexOf(searchText.toLowerCase());
  if (index >= 0) {
    const prefixText = displayText.slice(0, Math.max(0, index));
    const toBold = displayText.slice(index, index + searchText.length);
    const afterText = displayText.slice(index + searchText.length);
    return (
      <span style={{ whiteSpace: "pre-wrap" }}>
        {prefixText}
        <span className="text-bold">{toBold}</span>
        {afterText}
      </span>
    );
  } else {
    return <>{displayText}</>;
  }
};
