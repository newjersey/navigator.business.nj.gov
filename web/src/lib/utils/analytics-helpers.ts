import { getIsApplicableToFunctionByFieldName } from "@/lib/domain-logic/essentialQuestions";
import analytics from "@/lib/utils/analytics";
import { ABExperience, ProfileData } from "@businessnjgovnavigator/shared/";
import { OperatingPhaseId } from "@businessnjgovnavigator/shared/operatingPhase";
import { BusinessPersona, ForeignBusinessType } from "@businessnjgovnavigator/shared/profileData";
import { UserData } from "@businessnjgovnavigator/shared/userData";

type RegistrationProgress = "Not Started" | "Began Onboarding" | "Onboarded Guest" | "Fully Registered";

export const setRegistrationDimension = (status: RegistrationProgress) => {
  analytics.dimensions.registrationStatus(status);
};

export const setABExperienceDimension = (value: ABExperience) => {
  analytics.dimensions.abExperience(value);
};

export const setPhaseDimension = (value: OperatingPhaseId) => {
  analytics.dimensions.phase(getPhaseDimension(value));
};

export const phaseChangeAnalytics = ({
  oldUserData,
  newUserData,
}: {
  oldUserData: UserData;
  newUserData: UserData;
}) => {
  if (oldUserData.profileData.operatingPhase === newUserData.profileData.operatingPhase) {
    return;
  } else if (
    oldUserData.profileData.operatingPhase === "NEEDS_TO_FORM" &&
    newUserData.profileData.operatingPhase === "NEEDS_TO_REGISTER_FOR_TAXES"
  ) {
    analytics.event.roadmap_dashboard.arrive.progress_to_needs_to_register_phase();
  } else if (
    oldUserData.profileData.operatingPhase === "NEEDS_TO_REGISTER_FOR_TAXES" &&
    newUserData.profileData.operatingPhase === "FORMED_AND_REGISTERED"
  ) {
    analytics.event.roadmap_dashboard.arrive.progress_to_formed_and_registered_phase();
  } else if (
    oldUserData.profileData.operatingPhase === "FORMED_AND_REGISTERED" &&
    newUserData.profileData.operatingPhase === "UP_AND_RUNNING"
  ) {
    analytics.event.roadmap_dashboard.arrive.progress_to_up_and_running_phase();
  }
};

export const setAnalyticsDimensions = (profileData: ProfileData): void => {
  analytics.dimensions.industry(profileData.industryId);
  analytics.dimensions.municipality(profileData.municipality?.displayName);
  analytics.dimensions.legalStructure(profileData.legalStructureId);
  analytics.dimensions.homeBasedBusiness(profileData.homeBasedBusiness ? "true" : "false");
  analytics.dimensions.persona(getPersonaDimension(profileData.businessPersona));
  analytics.dimensions.naicsCode(profileData.naicsCode);
  analytics.dimensions.phase(getPhaseDimension(profileData.operatingPhase));
  analytics.dimensions.subPersona(getSubPersonaDimension(profileData.foreignBusinessType));
};

const getPhaseDimension = (phase: OperatingPhaseId) => {
  switch (phase) {
    case "GUEST_MODE":
      return "Guest Mode Needs to Form";
    case "GUEST_MODE_OWNING":
      return "Guest Mode Up and Running";
    case "NEEDS_TO_FORM":
      return "Needs to Form";
    case "NEEDS_TO_REGISTER_FOR_TAXES":
      return "Needs to Register";
    case "FORMED_AND_REGISTERED":
      return "Formed and Registered";
    case "UP_AND_RUNNING":
      return "Up and Running";
    case "UP_AND_RUNNING_OWNING":
      return "Up and Running";
    default:
      return phase;
  }
};

const getPersonaDimension = (persona: BusinessPersona): string => {
  switch (persona) {
    case "STARTING":
      return "Prospective";
    case "OWNING":
      return "Existing";
    case "FOREIGN":
      return "Foreign Prospective";
    default:
      return "";
  }
};

const getSubPersonaDimension = (type: ForeignBusinessType): string => {
  switch (type) {
    case "REMOTE_WORKER":
      return "Remote Worker";
    case "REMOTE_SELLER":
      return "Remote Seller";
    case "NEXUS":
      return "Nexus Business";
    case "NONE":
      return "None of the Above";
    default:
      return "";
  }
};

export const sendOnboardingOnSubmitEvents = (newProfileData: ProfileData, pageName?: string): void => {
  if (pageName === "industry-page") {
    if (getIsApplicableToFunctionByFieldName("cannabisLicenseType")(newProfileData.industryId)) {
      if (newProfileData.cannabisLicenseType === "CONDITIONAL") {
        analytics.event.onboarding_cannabis_question.submit.conditional_cannabis_license();
      } else if (newProfileData.cannabisLicenseType === "ANNUAL") {
        analytics.event.onboarding_cannabis_question.submit.annual_cannabis_license();
      }
    }

    if (getIsApplicableToFunctionByFieldName("requiresCpa")(newProfileData.industryId)) {
      if (newProfileData.requiresCpa) {
        analytics.event.onboarding_cpa_question.submit.yes_i_offer_public_accounting();
      } else {
        analytics.event.onboarding_cpa_question.submit.no_i_dont_offer_public_accounting();
      }
    }

    if (getIsApplicableToFunctionByFieldName("interstateMoving")(newProfileData.industryId)) {
      if (newProfileData.interstateMoving) {
        analytics.event.onboarding_moving_company_question.submit.yes_moving_across_state_lines();
      } else {
        analytics.event.onboarding_moving_company_question.submit.no_moving_across_state_lines();
      }
    }

    if (getIsApplicableToFunctionByFieldName("interstateLogistics")(newProfileData.industryId)) {
      if (newProfileData.interstateLogistics) {
        analytics.event.onboarding_logistics_business_question.submit.yes_moving_across_state_lines();
      } else {
        analytics.event.onboarding_logistics_business_question.submit.no_not_moving_across_state_lines();
      }
    }

    if (getIsApplicableToFunctionByFieldName("carService")(newProfileData.industryId)) {
      switch (newProfileData.carService) {
        case "STANDARD": {
          analytics.event.onboarding_car_service_question.submit.taxi_size();
          break;
        }
        case "HIGH_CAPACITY": {
          analytics.event.onboarding_car_service_question.submit.large_size();
          break;
        }
        case "BOTH": {
          analytics.event.onboarding_car_service_question.submit.both_sizes();
          break;
        }
      }
    }

    if (getIsApplicableToFunctionByFieldName("isChildcareForSixOrMore")(newProfileData.industryId)) {
      if (newProfileData.isChildcareForSixOrMore) {
        analytics.event.onboarding_childcare_business_question.submit.yes_more_than_6_children();
      } else {
        analytics.event.onboarding_childcare_business_question.submit.no_5_or_fewer_children();
      }
    }

    if (getIsApplicableToFunctionByFieldName("liquorLicense")(newProfileData.industryId)) {
      if (newProfileData.liquorLicense) {
        analytics.event.onboarding_liquor_question.submit.yes_require_liquor_license();
      } else {
        analytics.event.onboarding_liquor_question.submit.no_dont_require_liquor_license();
      }
    }
  }
};
