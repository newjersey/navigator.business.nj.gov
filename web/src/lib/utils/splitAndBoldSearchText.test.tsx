import { splitAndBoldSearchText } from "@/lib/utils/splitAndBoldSearchText";
import { render, screen } from "@testing-library/react";

describe("splitAndBoldSearchText", () => {
  it("properly displays the found text as bold", () => {
    render(splitAndBoldSearchText("food truck", "truck"));
    expect(screen.getByText("food")).toBeInTheDocument();
    expect(screen.getByTestId("span-bold")).toHaveTextContent("truck");
  });

  it("properly displays the text as not bold when not found", () => {
    render(splitAndBoldSearchText("food truck", "firm"));
    expect(screen.queryByTestId("span-bold")).not.toBeInTheDocument();
    expect(screen.getByText("food truck")).toBeInTheDocument();
  });
});
