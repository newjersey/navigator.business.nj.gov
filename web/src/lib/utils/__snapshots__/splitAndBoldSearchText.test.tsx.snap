// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splitAndBoldSearchText properly displays the found text as bold 1`] = `
<span
  style={
    {
      "whiteSpace": "pre-wrap",
    }
  }
>
  food 
  <span
    className="text-bold"
  >
    truck
  </span>
  
</span>
`;

exports[`splitAndBoldSearchText properly displays the text as not bold when not found 1`] = `
<React.Fragment>
  food truck
</React.Fragment>
`;
